saq/queue/postgres.py

```diff
@@ -0,0 +1,334 @@
+            Otherwise, the pool will have no limit.
+    """
+
+    @classmethod
+    def from_url(cls: type[PostgresQueue], url: str, **kwargs: t.Any) -> PostgresQueue:
+        """Create a queue from a postgres url."""
+        return cls(AsyncNullConnectionPool(url, open=False), **kwargs)
+
+    def __init__(
+        self,
+        pool: AsyncConnectionPool,
+        name: str = "default",
+        dump: DumpType | None = None,
+        load: LoadType | None = None,
+        max_size: int = 5,
+    ) -> None:
+        super().__init__(name=name, dump=dump, load=load)
+
+        self.pool = pool
+        self.max_size = max_size
+
+    async def connect(self) -> None:
+        await self.pool.open()
+        await self.pool.resize(min_size=0, max_size=self.max_size)
+        async with self.pool.connection() as conn, conn.cursor() as cursor:
+            await cursor.execute(CREATE_JOBS_TABLE)
+
+    def job_id(self, job_key: str) -> str:
+        return job_key
+
+    async def disconnect(self) -> None:
+        await self.pool.close()
+
+    async def info(
+        self, jobs: bool = False, offset: int = 0, limit: int = 10
+    ) -> QueueInfo:
+        return {
+            "workers": {},
+            "name": self.name,
+            "queued": 0,
+            "active": 0,
+            "scheduled": 0,
+            "jobs": [],
+        }
+
+    async def stats(self, ttl: int = 60) -> QueueStats:
+        return {
+            "complete": self.complete,
+            "failed": self.failed,
+            "retried": self.retried,
+            "aborted": self.aborted,
+            "uptime": now() - self.started,
+        }
+
+    async def count(self, kind: CountKind) -> int:
+        status = None
+        if kind == "queued":
+            status = [Status.QUEUED]
+        if kind == "active":
+            status = [Status.ACTIVE]
+        if kind == "incomplete":
+            status = [Status.NEW, Status.DEFERRED, Status.QUEUED, Status.ACTIVE]
+
+        if status:
+            async with self.pool.connection() as conn, conn.cursor() as cursor:
+                await cursor.execute(
+                    f"""
+                    SELECT count(*) FROM {JOBS_TABLE}
+                    WHERE status IN %(status)s
+                    """,
+                    {"status": status},
+                )
+                result = await cursor.fetchone()
+                assert result
+                return result[0]
+        raise ValueError("Can't count unknown type {kind}")
+
+    async def schedule(self, lock: int = 1) -> t.Any: ...
+
+    async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[t.Any]:
+        return []
+
+    async def listen(
+        self,
+        job_keys: Iterable[str],
+        callback: ListenCallback,
+        timeout: float | None = 10,
+    ) -> None:
+        async with self.pool.connection() as conn:
+            await conn.set_autocommit(True)
```

Comment:
why do you manually neet to set this?

Reply:
Due to the way notifications are implemented in Postgres, if I don't set this, the LISTEN will be in a transaction and won't receive any notifications until the transaction ends.